### PR TITLE
refactor: migrate MyLegacyViewManager to kotlin

### PR DESCRIPTION
**Pull request by imalgrab**
## Summary:

Migrate MyLegacyViewManager to Kotlin (#38825)

## Changelog:

[INTERNAL] [CHANGED]  - Moved MyLegacyViewManager to Kotlin

## Test Plan:

Run `yarn && yarn android`

### CODE REVIEW

**Review by cortinico — CHANGES_REQUESTED** [review]
Hey @imalgrab 
Thanks for sending this over.

Your code is not compiling though, have you verified that you're able to run RN Tester? You can do so by running `yarn && yarn android` from the root.

Please make sure the `test_android` job is green and then we can review this PR.

**Comment by imalgrab** [comment]
Hey @cortinico 
I was trying to fix the RN Tester but turns out it fails even on `main` branch for me. I guess it's because of my environment setup, however I cannot find out what's the cause.
First, I used JDK 11 and I got an error to use 17 instead, after that I have another error:
```
> ~/Pr/react-native main ❯ yarn && yarn android
...
> Task :packages:react-native:ReactAndroid:hermes-engine:configureBuildForHermes FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
```
Do you maybe have any ideas what could be the cause?
```
~/Projects/react-native main ❯ ./gradlew -v                                                                                    48s 18:47:03

------------------------------------------------------------
Gradle 8.2.1
------------------------------------------------------------

Build time:   2023-07-10 12:12:35 UTC
Revision:     a38ec64d3c4612da9083cc506a1ccb212afeecaa

Kotlin:       1.8.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.8 (Azul Systems, Inc. 17.0.8+7-LTS)
OS:           Mac OS X 13.4 aarch64
```

**Comment by BogiKay** [comment]
@imalgrab Looks like updating gradle will solve the problem. If you have installed OpenJDK Zulu11 like [docs](https://reactnative.dev/docs/environment-setup?platform=android) says, you can just install newer version of distribution like 17 or 20. They come with gradle 9.

**Comment by imalgrab** [comment]
Hey @BogiKay thanks for taking your time to help me! Turns out solution mentioned in this issue https://github.com/facebook/react-native/issues/34994#issuecomment-1285981569 solved the problem for me so I'll try to fix the `test_android` job now

**Comment by imalgrab** [comment]
Hi @cortinico 
Could you please help me with `analyze_pr` job? I see linters fail

**Comment by cortinico** [comment]
@imalgrab could you rebase on top of `main`?

**Comment by imalgrab** [comment]
Hi @cortinico 
Thanks, rebasing the branch on top of `main` did the job for `analyze_pr` job.
However, it struggles with `test_e2e_android`. I see it fails because of a timeout probably, when I tried to verify on my end the `yarn android` script build successfully and then immediately app closes. Do you have any suggestions what can be done? I don't see any problems when running `test-android` also.

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    super.setOpacity(view, opacity)
+  }
+
+  @ReactProp(name = ViewProps.COLOR)
+  fun setColor(view: MyNativeView, color: String?) {
+    view.setBackgroundColor(Color.parseColor(color))
+  }
+
+  @ReactProp(name = "cornerRadius")
+  fun setCornerRadius(view: MyNativeView, cornerRadius: Float?) {
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:55]:
```suggestion
  fun setCornerRadius(view: MyNativeView, cornerRadius: Float) {
```

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    return view
+  }
+
+  @ReactProp(name = ViewProps.OPACITY, defaultFloat = 1f)
+  override fun setOpacity(view: MyNativeView, opacity: Float) {
+    super.setOpacity(view, opacity)
+  }
+
+  @ReactProp(name = ViewProps.COLOR)
+  fun setColor(view: MyNativeView, color: String?) {
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:50]:
```suggestion
  fun setColor(view: MyNativeView, color: String) {
```

**Review by cortinico — CHANGES_REQUESTED** [review]
The failure is actually valid, RN Tester is insta-crashing for me if you `yarn android` and I'm glad that we have E2E tests now in place that captured this before we merged this change as this would have been a pain to track down afterwards (cc @kelset @cipolleschi @thymikee fyi).

@imalgrab please apply my suggestions and the tests should be green afterwards

**Comment by imalgrab** [comment]
@cortinico I applied the requested changes, thanks! Android tests and e2e tests run correctly but `test_ios_template-OldArch-Debug-WithoutFlipper-Hermes-DynamicFrameworks` could you help here, please?

**Comment by cortinico** [comment]
@imalgrab that job is probably flaky, I've re-executed it to see if it turns red.

**Comment by imalgrab** [comment]
@cortinico you were right, now it's executed as expected

**Review by cortinico — APPROVED** [review]

**Comment by imalgrab** [comment]
@cortinico could you please assist me with Internal Linter job?

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  companion object {
+    const val REACT_CLASS = "RNTMyLegacyNativeView"
+    const val COMMAND_CHANGE_BACKGROUND_COLOR = 42
+  }
+
+  private val mCallerContext: ReactApplicationContext = reactContext
+
+  override fun getName(): String {
+    return REACT_CLASS
+  }
```
Comment (rshest) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:36]:
```suggestion
  override fun getName(): String = REACT_CLASS
```

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ */
+@ReactModule(name = MyLegacyViewManager.REACT_CLASS)
+internal class MyLegacyViewManager(reactContext: ReactApplicationContext) :
+  SimpleViewManager<MyNativeView>() {
+  companion object {
+    const val REACT_CLASS = "RNTMyLegacyNativeView"
+    const val COMMAND_CHANGE_BACKGROUND_COLOR = 42
+  }
+
+  private val mCallerContext: ReactApplicationContext = reactContext
```
Comment (rshest) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:32]: Please avoid using Hungarian notation with Kotlin (i.e. 'callerContext' here).

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+ * Legacy View manager (non Fabric compatible) for {@link MyNativeView}
+ * components.
+ */
+@ReactModule(name = MyLegacyViewManager.REACT_CLASS)
+internal class MyLegacyViewManager(reactContext: ReactApplicationContext) :
+  SimpleViewManager<MyNativeView>() {
+  companion object {
+    const val REACT_CLASS = "RNTMyLegacyNativeView"
+    const val COMMAND_CHANGE_BACKGROUND_COLOR = 42
+  }
```
Comment (rshest) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:30]: Can we have the companion object at the end of the class?

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  @ReactProp(name = "cornerRadius")
+  fun setCornerRadius(view: MyNativeView, cornerRadius: Float) {
+    if (cornerRadius !== null) {
+      view.setCornerRadius(cornerRadius)
+    }
+  }
+
+  override fun getExportedViewConstants(): Map<String, Any> {
+    return mapOf("PI" to 3.14)
+  }
```
Comment (rshest) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:63]:
```suggestion
  override fun getExportedViewConstants(): Map<String, Any> = mapOf("PI" to 3.14)
```

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+  override fun getExportedCustomBubblingEventTypeConstants(): Map<String, Any> {
+    return MapBuilder.builder<String, Any>()
+      .put(
```
Comment (rshest) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:67]:
```suggestion
    return mapOf(
```

**Review by rshest — CHANGES_REQUESTED** [review]

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+  override fun getName(): String {
+    return REACT_CLASS
+  }
+
+  override fun createViewInstance(reactContext: ThemedReactContext): MyNativeView {
+    val view: MyNativeView = MyNativeView(reactContext)
+    view.setBackgroundColor(Color.RED)
+    return view
+  }
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:42]:
```suggestion
  override fun createViewInstance(reactContext: ThemedReactContext): MyNativeView =
    MyNativeView(reactContext).apply {
      setBackgroundColor(Color.RED)
    }
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Please follow up on @rshest and my comments here and then we can merge this

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt (hunk 7 above)
Comment (imalgrab) [reply packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyLegacyViewManager.kt:67]: Hey @rshest thanks for the help!
Unfortunately, RNTesterApp does not open for neither
```
    return mapOf(
      "onColorChanged" to
        MapBuilder.of(
          "phasedRegistrationNames",
          MapBuilder.of(
            "bubbled",
            "onColorChanged",
            "captured",
            "onColorChangedCapture"
          )
        )
    )
```
nor
```
    return mapOf(
      "onColorChanged" to
        mapOf(
          "phasedRegistrationNames" to
            mapOf(
              "bubbled" to
                "onColorChanged",
              "captured" to
                "onColorChangedCapture"
            )
        )
    )
```
I applied other requested changes without any problem and the app runs correctly.

**Pull request closed without merging**